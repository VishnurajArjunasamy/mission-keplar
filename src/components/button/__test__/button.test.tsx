import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button from "../button";

describe("Button Component", () => {
  it("renders the button with the provided label", () => {
    const { asFragment } = render(<Button label="Click Me" />);

    expect(screen.getByText("Click Me")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies the default type 'button' if no type is provided", () => {
    const { asFragment } = render(<Button label="Click Me" />);

    const button = screen.getByText("Click Me");
    expect(button).toHaveAttribute("type", "button");

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies the provided type attribute", () => {
    const { asFragment } = render(<Button label="Submit" type="submit" />);

    const button = screen.getByText("Submit");
    expect(button).toHaveAttribute("type", "submit");

    expect(asFragment()).toMatchSnapshot();
  });

  it("calls the onClick function when clicked", () => {
    const handleClick = jest.fn();

    const { asFragment } = render(
      <Button label="Click Me" onClick={handleClick} />
    );

    const button = screen.getByText("Click Me");
    fireEvent.click(button);

    expect(handleClick).toHaveBeenCalledTimes(1);

    expect(asFragment()).toMatchSnapshot();
  });
});
