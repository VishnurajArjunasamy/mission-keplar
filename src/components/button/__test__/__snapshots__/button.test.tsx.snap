// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component applies the default type 'button' if no type is provided 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button Component applies the provided type attribute 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="submit"
  >
    Submit
  </button>
</DocumentFragment>
`;

exports[`Button Component calls the onClick function when clicked 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Button Component renders the button with the provided label 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;
